### PR TITLE
navigate through the awesomebar with keys

### LINKED ISSUE #225: Add support for "arrow down" to awesomebar results
Currently you have to click.

cc @paulrouget

### PR DESCRIPTION
**Pull request by paulrouget**
Fix #225

### CODE REVIEW

**Comment by paulrouget** [comment]
@Gozala r?

File: src/browser/awesomebar.js [hunk 1]
```diff
@@ -11,42 +11,15 @@ define((require, exports, module) => {
...
-      let href = history[i].get('href');
-      let text = history[i].get('text');
-      suggestions.push(DOM.p({
-        className: 'suggestion history',
-        key: 'suggestionHistory' + i,
-        href,
-      }, text));
-    }
+    const selectedIndex = suggestionsCursor.get('selectedIndex');
+    const list = suggestionsCursor.get('list').toJSON();
```
Comment (Gozala) [inline src/browser/awesomebar.js]: Nit: You can call `.map()` on Immutable list, no need to transform to JSON.

File: src/browser/navigation-panel.js [hunk 2]
```diff
@@ -45,17 +46,76 @@ define((require, exports, module) => {
...
+    const size = suggestionsCursor.get('list').size;
+    let value = suggestionsCursor.get('selectedIndex') + delta;
+    if (value >= size) {
+      value = -1;
+    }
+    value = Math.max(-1, value);
+    suggestionsCursor.set('selectedIndex', value);
+  }
+
+  const onInputKeyDown = ({event, inputCursor, webViewerCursor, suggestionsCursor}) => {
```
Comment (Gozala) [inline src/browser/navigation-panel.js]: Can you please use `KeyBindings` instead as it does basically what you do here.

**Comment by Gozala** [comment]
As a general note, you are abusing `fromJS` in most cases you don't actually need it. Also use of Immutable (Map, List, Set) constructors tends to be faster.

**Comment by Gozala** [comment]
Overall this looks alright, but I would very much like to leverage existing `KeyBindings` for defining keybindings as it has more declarative interface and + all issues will be scoped in single location.

That being said I don't mind doing it as a followup pull.

**Pull request merged**